codex-rs/core/tests/suite/user_shell_cmd.rs

```diff
@@ -0,0 +1,119 @@
+#![cfg(unix)]
+
+use codex_core::ConversationManager;
+use codex_core::NewConversation;
+use codex_core::protocol::EventMsg;
+use codex_core::protocol::ExecCommandEndEvent;
+use codex_core::protocol::Op;
+use codex_core::protocol::TurnAbortReason;
+use core_test_support::load_default_config_for_test;
+use core_test_support::wait_for_event;
+use std::path::PathBuf;
+use tempfile::TempDir;
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
```

Comment:
Please omit `worker_threads` unless you have a reason to believe you need them.

Empirically, tests that only pass with more than one worker thread imply bugs in our code, so I would prefer to try to catch them.